compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -528,9 +528,7 @@ fn ty_known_to_outlive<'tcx>(
         let sup_type = ty;
         let sub_region = region;
 
-        let origin = SubregionOrigin::from_obligation_cause(&cause, || {
-            infer::RelateParamBound(cause.span, sup_type, None)
-        });
+        let origin = infer::RelateRegionParamBound(cause.span);
```

Comment:
This and the one below are drive-by changes. These shouldn't ever be displayed anyways. But wanted to stop using `from_obligation_cause` (was potentially going a different direction for this PR).